documentation/README_interface.md

```diff
@@ -0,0 +1,131 @@
+If a valid CRC is not transmitted, the command will be ignored.
+
+
+## Sensor message (MISO line)
+A sensor message is a message transmitted from the µDriver to the Master.
+
+Status | Time Stamp | Positions | Velocities | Iq | Coils Resistances | ADC | last cmd Index | /CRC
+--- | --- | --- | --- | --- | --- | --- | --- | ---
+16bits | 16bits | 2\*32bits | 2\*16bits | 2\*16bits | 2\*16bits | 2\*16bits | 16bits | 32bits
+
+> **Important note:** Compared to the behavior of the CAN interface the data are sampled at a rate imposed by the master. In the current implementation, the data sensor message is updated internally at a rate of about 1.5kHz.
+
+>The data read can then be outdated up to 0.66ms. Doing finite differences on position data can lead to sampling artifacts.  Later implementation could include a Timestamp field if needed.
+
+
+### Status:
+A status register is used to get the state of the motor board, the status of calibration, the error codes etc...
+
+ 15 | 14 | 13 | 12 | 11 | 10 | 9 | 8 | 7 | 6 | 5 | 4 | 3 | 2 | 1 | 0 
+--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- 
+ SE | M1E | M1R | M2E | M2R | IDX1D | IDX2D | IDX1T |IDX2T | - | - | -  | Error code | Error code | Error code | Error code 
+
+
+| | | |
+|---|---|---|
+| SE <15> (System Enabled ) | 0: System is disabled<br>1: System is enabled | |
+| M1E <14> (Motor 1 Enabled) | 0: Motor 1 is disabled<br>1: Motor 1 is enabled | Motor might be disabled due to user command, or due to an error. |
+| M1R <13> (Motor 1 Ready) | 0: Motor 1 is not ready<br>1: Motor 1 is ready | |
+| M2E <12> (Motor 2 Enable) | 0: Motor 2 is disabled<br>1: Motor 2 is enabled | Motor might not be ready due to calibration running.|
+| M2R <11> (Motor 2 Ready) | 0: Motor 2 is not ready<br>1: Motor 2 is ready | |
+| IDX1D <10> (Index 1 Detected) | 0: No encoder index has been detected yet<br>1: At least one encoder index has been detected | This bit is important to check before setting EIOC in the Mode word. |
+| IDX2D <9> (Index 2 Detected)|  0: No encoder index has been detected yet<br>1: At least one encoder index has been detected | This bit is important to check before setting EIOC in the Mode word. |
+| IDX1T <8> (Index 1 Toggle) | Flips each time an encoder index has been detected | |
+| IDX2T <7> (Index 2 Toggle) | Flips each time an encoder index has been detected | |
+| - <6:4> | Not used | Reading this bits has no meaning.
+| ErrorCode <3:0> | See CAN error code | Error code are now inherited from BLMC CAN Interface
+
+### /CRC:
+
+The integrity of the messages is checked with a CRC32 according to (https://github.com/gcc-mirror/gcc/blob/master/libiberty/crc32.c) witch output need to be complemented (XOR with 0xFFFF)
+CRC value must be checked before interpreting any of the sensor message data.
+
+**Why a complemeted CRC?**
```

Comment:
This is not actually implemented... We need to implement it or remove it from the doc. I would suggest to remove it for now